src/models/link/metadata.rs

```diff
@@ -98,200 +185,94 @@ impl LinkMetadataBuilder {
         self
     }
 
-    pub fn build(self) -> Result<LinkMetadata> {
-        LinkMetadata::new(
-            self.name,
+    pub(crate) fn build(self) -> Result<MetadataFlatten> {
+        MetadataFlatten::new(
+            self.builder_id,
+            self.build_invocation_id,
+            self.build_type,
+            self.config_uri,
+            self.config_filename,
+            self.config_digest,
             self.materials,
             self.products,
-            self.env,
+            self.parameters,
+            self.environment,
             self.byproducts,
             self.command,
         )
     }
 
     /// Construct a new `Metablock<D, LinkMetadata>`.
-    pub fn signed<D>(self, private_key: &PrivateKey) -> Result<Metablock<D, LinkMetadata>>
+    pub fn signed<D>(
+        self,
+        format: MetaFormat,
+        private_key: &PrivateKey,
+    ) -> Result<Metablock<D, StatementWrapper>>
     where
         D: DataInterchange,
     {
-        Metablock::new(&self.build()?, Some(private_key))
+        let flatten = self.build()?;
+        let wrapper = format.build_layout(&flatten)?;
+        Metablock::new(&wrapper, Some(private_key))
     }
 
     /// Construct a new `Metablock<D, LinkMetadata>`.
-    pub fn unsigned<D>(self) -> Result<Metablock<D, LinkMetadata>>
+    pub fn unsigned<D>(self, format: MetaFormat) -> Result<Metablock<D, StatementWrapper>>
     where
         D: DataInterchange,
     {
-        Metablock::new(&self.build()?, None)
+        let flatten = self.build()?;
+        let wrapper = format.build_layout(&flatten)?;
+        Metablock::new(&wrapper, None)
     }
 }
 
 /// link metadata
-#[derive(Debug, Clone, PartialEq)]
-pub struct LinkMetadata {
-    name: String,
+#[derive(Debug, Clone, PartialEq, Eq, Getters)]
+pub struct MetadataFlatten {
```

Comment:
migrate from `LinkMetadata` into `MetadataFlatten`, the latter acts as a flat field database to generating any attestation.